lib/at_host/at_host.c

```diff
@@ -214,7 +216,7 @@ static void isr(const struct device *dev, void *user_data)
 	 * Check that we are not sending data (buffer must be preserved then),
 	 * and that a new character is available before handling each character
 	 */
-	while ((!k_work_pending(&cmd_send_work)) &&
+	while ((!at_buf_busy) &&
```

Comment:
Out-of-scope: I wonder if it would have been better to offload this ```while()``` waiting to a thread instead of having it in an ISR callback. Is there a risk that we get stuck here for longer durations?

Reply:
It is actually not a waiting loop; `uart_rx_handler` is called within it. The prerequisites for reading a new character is
* A command is not being processed (`at_buf_busy` false)
* New character is available (`uart_fifo_read(dev, &character, 1)` returns 1). This will never keep returning 1 infinitely, AFAIK.

Once any of those are not met, the loop is terminated. This happens if a termination is detected in uart_rx_handler (that will set `at_buf_busy`) or there is no more data from UART.

Reply:
I see. Thank you for clarifying.